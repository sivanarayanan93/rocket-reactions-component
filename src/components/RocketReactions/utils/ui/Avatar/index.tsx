import styled from "styled-components";
import { getDefaultAvataorURL } from "../../app-utils";

const UiAvatar = styled.div`  
  display: inherit;
  img {
    width: 18px;
    height: 18px;
    border-radius: 50%;
  }
`

const defaultAvatar = getDefaultAvataorURL();

const Avatar = ({ url }: { url: string}) => {
  return (
    <UiAvatar>
      <img alt="Avatar" src={url || defaultAvatar} />
    </UiAvatar>
  )
}

export default Avatar;
