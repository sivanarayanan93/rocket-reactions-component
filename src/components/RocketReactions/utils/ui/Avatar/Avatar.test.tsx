import { render } from '@testing-library/react';
import Avatar from './index';

describe('Avatar component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Avatar url ="https://dummuy.com/image.png"/>);
    expect(asFragment()).toMatchSnapshot();
  });
});