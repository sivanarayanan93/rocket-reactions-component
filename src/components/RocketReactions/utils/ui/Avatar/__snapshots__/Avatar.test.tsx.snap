// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM fFeCqH"
  >
    <img
      alt="Avatar"
      src="https://dummuy.com/image.png"
    />
  </div>
</DocumentFragment>
`;
